pymeasure/instruments/cisystems/sr200n.py

```diff
@@ -0,0 +1,181 @@
+
+    temp_is_stable = CommonBase.measurement(
+        TEMP_STABLE,
+        """Boolean parameter to check if the temperature has stabilized.""",
+        cast=bool,
+    )
+    
+    set_point_min = CommonBase.measurement(
+        SET_POINT_MIN,
+        """Float representing the minimum temperature set point.""",
+    )
+    
+    set_point_max = CommonBase.measurement(
+        SET_POINT_MAX,
+        """Float representing the maximum temperature set point.""",
+    )
+    
+    set_point = CommonBase.control(
+        SET_POINT_GET, SET_POINT_SET,
+        """Float representing the blackbody temperature set point.""",
+    )
+
+    # - chopper parameters ---------------------------------- #
+    chopper_frequency = CommonBase.control(
+        CHOPPER_FREQ_GET, CHOPPER_FREQ_SET,
+        """Float representing the frequency of the internal chopper."""
+    )
+
+    chopper_position = CommonBase.setting(
+        CHOPPER_POS,
+        """Integer for manual control of the chopper (0 for closed, 1 for open)."""
+    )
+
+    # - internal parameters for constructing and reading messages - #
+    _header = b'\xaa\x01' 
+    _read_dict = {
+        TEMP: (14, float),
+        TEMP_STABLE: (11, bool),
+        SET_POINT_MIN: (14, float),
+        SET_POINT_MAX: (14, float),
+        SET_POINT_GET: (14, float),
+        CHOPPER_FREQ_GET: (14, float),
+    }
+    _read_count = 0
+    _read_type = None
+
+    def __init__(self, adapter, name='SR200N Blackbody', **kwargs):
+        super().__init__(
+            adapter,
+            name,
+            asrl={
+                'baud_rate': 115200,
+            },
+            **kwargs
+        )
+
+    @staticmethod
+    def add_bytes(byte1, byte2):
+        """ Static method to implement the addition of two bytes with overflow. Used for checksum
+        calculation.
+
+        :param byte1: First byte object to add.
+        :param byte2: Second byte object to add.
+        """
+        int_value = (int.from_bytes(byte1, 'big') + int.from_bytes(byte2, 'big')) & 0xff
+        byte_value = int_value.to_bytes(length=4, byteorder='big')
+        return byte_value        
+
+    def calc_checksum(self, add):
+        """ Calculate a checksum for a string of bytes.
+
+        :param add: Byte string to calculate a checksum on.
+        """
+        sum_bytes = bytes.fromhex('00')
+        for i in range(len(add)):
+            b = add[i:i+1]
+            sum_bytes = self.add_bytes(sum_bytes, b)
+        b255 = bytes.fromhex('ff')
+        cs = int.from_bytes(b255, 'big') - int.from_bytes(sum_bytes, 'big') + 1
+        
+        return cs.to_bytes(length=4, byteorder='big')
+
+    def write(self, command, **kwargs):
+        """ Override the base write method to construct the checksum and use `write_bytes()`.
+
+        :param command: hexadecimal command as a string.
+        """
+        # - split the command string on spaces, get the command value if present - #
+        cmd_split = command.split(' ')
```

Comment:
In the hcp TC038D I used a comma as separator, which is more common, I'd think.

Here's the code, maybe it serves you as inspiration:

https://github.com/pymeasure/pymeasure/blob/f3d8f84c3e84b9de368f392a43accaac6423a602/pymeasure/instruments/hcp/tc038d.py#L76-L105